### PR TITLE
Fix even more edge cases for database migration.

### PR DESCRIPTION
**Pull request by dbrant**
* Make the test parameterized, and give it versions `19` and `22` to test with.
* Roll the tests into a single `@Test` function, which makes it faster and doesn't lose any coverage.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/database/AppDatabase.kt [hunk 1]
```diff
@@ -86,8 +87,16 @@ abstract class AppDatabase : RoomDatabase() {
...
+                try {
+                    database.query("SELECT * FROM sqlite_master WHERE type='table' AND name='talkpageseen'").use {
+                        if (it.count > 0) {
+                            database.execSQL("INSERT OR REPLACE INTO TalkPageSeen_temp (sha) SELECT sha FROM talkpageseen")
+                            database.execSQL("DROP TABLE talkpageseen")
+                        }
+                    }
+                } catch (e: Exception) {
+                    // ignore further errors
+                }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/database/AppDatabase.kt:99]: This makes sense to me, and would it be possible (or necessary?) to apply this logic to all other tables?

Comment (dbrant) [reply app/src/main/java/org/wikipedia/database/AppDatabase.kt:99]: @cooltey I'm not so sure...  I put these exception handlers only for the small remainder of crashes that I can't explain.  There haven't been any crashes in any of the other conversion statements, and if there is a problem in those statements I'd really like to see it via crash report. (Otherwise we risk data loss for the user.)

Comment (cooltey) [reply app/src/main/java/org/wikipedia/database/AppDatabase.kt:99]: Got it, that makes sense to me. It looks good to me so far, @sharvaniharan please review and merge if it looks good to you too.
